### PR TITLE
Optimized fibonacci

### PR DESCRIPTION
**Pull request by FantassyLoop**
Resolves Issue #<48>

### Description
Made an array to store all the previous fibonacci numbers so that there is no unnecessary computation.

### CODE REVIEW

**Comment by FantassyLoop** [comment]
#48

File: algorithms/48/fib.cpp [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
 int main() {
     int n;
     cout << "Enter a number: ";
     cin >> n;
-    cout << "The " << n << "th fibonacci number is " << fib(n) << endl;
+	long long fib[n+1];
+	fib[1]=0;
+	fib[2]=1;
+	for(int i=3;i<=n;i++)
+		fib[i]=fib[i-1]+fib[i-2];
```
Comment (shanthanu9) [inline algorithms/48/fib.cpp]: You have removed the fib function entirely here. It's better to keep the ```fib``` function and implement your optimization in ```fib```, this leads to cleaner and modular code.
Also you have inconsistant indentation here, please maintain the same indentation as the original file.

Comment (shanthanu9) [inline algorithms/48/fib.cpp]: In the original code, 
```
fib(0) -> 0
fib(1) -> 1
fib(2) -> 1
fib(3) -> 2
```

However, in your code it is,
```
fib(0) -> 0
fib(1) -> 0
fib(2) -> 1
fib(3) -> 1
```

Could you change your code to match the same output as the original one?

File: algorithms/48/fib.cpp [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-    if(n == 1) return 1;
-    return fib(n-1)+fib(n-2);
-}
 
 int main() {
     int n;
     cout << "Enter a number: ";
     cin >> n;
-    cout << "The " << n << "th fibonacci number is " << fib(n) << endl;
+	long long fib[n+1];
```
Comment (shanthanu9) [inline algorithms/48/fib.cpp]: Do you need to store all the fibonacci numbers less than the ```nth``` fibonacci number to compute it?

**Comment by shanthanu9** [comment]
Please go through the suggested changes above.

**Comment by shanthanu9** [comment]
Hey is there any confusion regarding my comments in your review since you haven't yet responded? Feel free to ask me any doubt you have.

File: algorithms/48/fib.cpp (hunk 2 above)
Comment (FantassyLoop) [reply algorithms/48/fib.cpp]: This was done to reduce computational load

File: algorithms/48/fib.cpp (hunk 1 above)
Comment (FantassyLoop) [reply algorithms/48/fib.cpp]: Will do it soon

**Comment by SaurabhAgarwala** [comment]
Hacktoberfest 2019 ends. Thanks for contributing. Closing the PR.

**Pull request closed without merging**